crypto/keyring/keyring.go

```diff
@@ -842,26 +837,166 @@ func (ks keystore) existsInDb(info Info) (bool, error) {
 	return false, nil
 }
 
-func (ks keystore) writeOfflineKey(name string, pub types.PubKey, algo hd.PubKeyType) (Info, error) {
-	info := newOfflineInfo(name, pub, algo)
-	err := ks.writeInfo(info)
+func (ks keystore) writeOfflineKey(name string, pk types.PubKey) (*Record, error) {
+	k, err := NewOfflineRecord(name, pk)
 	if err != nil {
 		return nil, err
 	}
 
-	return info, nil
+	return k, ks.writeRecord(k)
 }
 
-func (ks keystore) writeMultisigKey(name string, pub types.PubKey) (Info, error) {
-	info, err := NewMultiInfo(name, pub)
+// writeMultisigKey investigate where thisf function is called maybe remove it
+func (ks keystore) writeMultisigKey(name string, pk types.PubKey) (*Record, error) {
+	k, err := NewMultiRecord(name, pk)
 	if err != nil {
 		return nil, err
 	}
-	if err = ks.writeInfo(info); err != nil {
+
+	return k, ks.writeRecord(k)
+}
+
+func (ks keystore) MigrateAll() (bool, error) {
+	var migrated bool
```

Comment:
please move this line just above the for loop, and in between in return statements use explicit `false`.